src/OrchardCore.Modules/OrchardCore.Media/Recipes/MediaStep.cs

```diff
@@ -32,29 +28,46 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
 
             var model = context.Step.ToObject<MediaStepModel>();
 
-            foreach (JObject item in model.Files)
+            if (model.Files != null)
             {
-                var file = item.ToObject<MediaStepFile>();
-                using (var stream = new MemoryStream(Convert.FromBase64String(file.Base64)))
+                foreach (var file in model.Files)
                 {
-                    await _mediaFileStore.CreateFileFromStream(file.Path, stream, true);
+                    using (var stream = new MemoryStream(Convert.FromBase64String(file.Base64)))
+                    {
+                        await _mediaFileStore.CreateFileFromStream(file.Path, stream, true);
+                    }
+                }
+            }
+
+            if (model.Paths != null)
+            {
+                foreach (var path in model.Paths)
+                {
+                    var fileInfo = context.RecipeDescriptor.FileProvider.GetFileInfo(path);
+
+                    using (var stream = fileInfo.CreateReadStream())
+                    {
+                        await _mediaFileStore.CreateFileFromStream(path, stream, true);
+                    }
                 }
             }
         }
-    }
 
-    public class MediaStepModel
-    {
-        public JArray Files { get; set; }
-    }
+        private class MediaStepModel
```

Comment:
I am confused between MediaStepModel.Paths and MediaStepFile.Path. Could you add some documentation on the class and properties, as little as necessary to not be confused anymore.